lib/ruby_lsp/extension.rb

```diff
@@ -130,7 +130,7 @@ def create_hover_listener(emitter, message_queue); end
       overridable.params(
         emitter: EventEmitter,
         message_queue: Thread::Queue,
-      ).returns(T.nilable(Listener[T.nilable(Interface::DocumentSymbol)]))
+      ).returns(T.nilable(Listener[T::Array[Interface::DocumentSymbol]]))
```

Comment:
I wonder if we should use `Requests::DocumentSymbol::ResponseType` instead to prevent any inconsistencies again.

Reply:
Oh, okay. Yeah, that makes sense, you shouldn't have access to type variables outside of their scope.